test/e2e/apimachinery/custom_resource_definition.go

```diff
@@ -42,29 +51,153 @@ var _ = SIGDescribe("CustomResourceDefinition resources", func() {
+			crd := fixtures.NewRandomNameCustomResourceDefinition(v1beta1.ClusterScoped)
+			crd, err = fixtures.CreateNewCustomResourceDefinition(crd, apiExtensionClient, f.DynamicClient)
+			framework.ExpectNoError(err, "creating CustomResourceDefinition")
+			defer func() {
+				err = fixtures.DeleteCustomResourceDefinition(crd, apiExtensionClient)
+				framework.ExpectNoError(err, "deleting CustomResourceDefinition")
+			}()
+
+			selectorListOpts := metav1.ListOptions{LabelSelector: "e2e-list-test-uuid=" + testUUID}
+			list, err := apiExtensionClient.ApiextensionsV1beta1().CustomResourceDefinitions().List(selectorListOpts)
+			framework.ExpectNoError(err, "listing CustomResourceDefinitions")
+			gomega.Expect(len(list.Items)).To(gomega.Equal(testListSize))
+			for _, actual := range list.Items {
+				var expected *v1beta1.CustomResourceDefinition
+				for _, e := range crds {
+					if e.Name == actual.Name && e.Namespace == actual.Namespace {
+						expected = e
+					}
+				}
+				gomega.Expect(expected).To(gomega.Not(gomega.BeNil()))
+				if !equality.Semantic.DeepEqual(actual.Spec, expected.Spec) {
+					e2elog.Failf("Expected CustomResourceDefinition in list with name %s to match crd created with same name, but got different specs:\n%s",
+						actual.Name, diff.ObjectReflectDiff(expected.Spec, actual.Spec))
 				}
+			}
+
+			// Use delete collection to remove the CRDs
+			err = fixtures.DeleteCustomResourceDefinitions(selectorListOpts, apiExtensionClient)
+			framework.ExpectNoError(err, "deleting CustomResourceDefinitions")
+		})
+
+		/*
+			Release : v1.16
+			Testname: Custom Resource Definition, status sub-resource
+			Description: Create a API extension client, create a custom resource definition and then read, update and patch its status sub-resource. API server MUST be able to perform the operations against the status sub-resource.
+		*/
+		ginkgo.It("getting/updating/patching custom resource definition status sub-resource works ", func() {
+			config, err := framework.LoadConfig()
+			framework.ExpectNoError(err, "loading config")
+			apiExtensionClient, err := clientset.NewForConfig(config)
+			framework.ExpectNoError(err, "initializing apiExtensionClient")
+			dynamicClient, err := dynamic.NewForConfig(config)
+			framework.ExpectNoError(err, "initializing dynamic client")
+
+			// Create CRD and waits for the resource to be recognized and available.
+			crd := fixtures.NewRandomNameCustomResourceDefinition(v1beta1.ClusterScoped)
+			crd, err = fixtures.CreateNewCustomResourceDefinition(crd, apiExtensionClient, f.DynamicClient)
+			framework.ExpectNoError(err, "creating CustomResourceDefinition")
+			defer func() {
+				err = fixtures.DeleteCustomResourceDefinition(crd, apiExtensionClient)
+				framework.ExpectNoError(err, "deleting CustomResourceDefinition")
 			}()
+
+			// Use dynamic client to test status sub-resource since typed client does not expose status resource.
```

Comment:
really? can we not use https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/typed/apiextensions/v1beta1/customresourcedefinition.go#L42 and https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/typed/apiextensions/v1beta1/customresourcedefinition.go#L48 ?

Reply:
Oh duh. So I can do everything except the get. I'd prefer not do the get, but aipsnoop flagged it (https://github.com/kubernetes/kubernetes/issues/81192), so I guess I have leave that in?

Reply:
leaving it in is fine. low-value, but fine.